cluster-state-service/internal/features/e2etasksteps/common_steps.go

```diff
@@ -179,3 +233,96 @@ func startNTasks(numTasks int, startedBy string, ecsWrapper wrappers.ECSWrapper)
 		EcsTaskList = append(EcsTaskList, ecsTask)
 	}
 }
+
+func createInstanceProfile(iamWrapper wrappers.IAMWrapper) error {
+	assumeRolePolicy := `{
+		"Version": "2012-10-17",
+		"Statement": [
+		{
+		"Effect": "Allow",
+		"Principal": {
+			"Service": "ec2.amazonaws.com"
+		},
+		"Action": "sts:AssumeRole"
+		}
+		]
+	}`
+
+	err := iamWrapper.GetRole(&roleName)
+	if err != nil {
+		err = iamWrapper.CreateRole(&roleName, &assumeRolePolicy)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = iamWrapper.GetInstanceProfile(&instanceProfileName)
+	if err != nil {
+		err = iamWrapper.CreateInstanceProfile(&instanceProfileName)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = iamWrapper.AttachRolePolicy(&policyArn, &roleName)
+	if err != nil {
+		return err
+	}
+
+	err = iamWrapper.AddRoleToInstanceProfile(&roleName, &instanceProfileName)
+
+	time.Sleep(time.Duration(5) * time.Second)
+
+	return nil
+}
+
+func deleteInstanceProfile(iamWrapper wrappers.IAMWrapper) error {
+	err := iamWrapper.DetachRolePolicy(&policyArn, &roleName)
+	if err != nil {
+		return err
+	}
+
+	err = iamWrapper.RemoveRoleFromInstanceProfile(&roleName, &instanceProfileName)
+	if err != nil {
+		return err
+	}
+
+	err = iamWrapper.DeleteRole(&roleName)
+	if err != nil {
+		return err
+	}
+
+	err = iamWrapper.DeleteInstanceProfile(&instanceProfileName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func terminateAllContainerInstances(ec2Wrapper wrappers.EC2Wrapper, ecsWrapper wrappers.ECSWrapper, clusterName string) error {
+	instanceARNs, err := ecsWrapper.ListContainerInstances(clusterName)
+	if len(instanceARNs) == 0 {
+		return nil
+	}
+	err = ecsWrapper.DeregisterContainerInstances(&clusterName, instanceARNs)
+	if err != nil {
+		return errors.Wrapf(err, "Fail to deregister container instances '%v'.", instanceARNs)
+	}
+
+	instanceIDs := make([]*string, len(instanceARNs))
```

Comment:
Change this to
`instanceIDs := make([]*string, 0, len(instanceARNs))`
If you default a slice to say length of 10, the first element that gets appended goes into the 11th spot, not the 1st (https://blog.golang.org/go-slices-usage-and-internals).